beacon_node/network/src/service.rs

```diff
@@ -40,48 +42,42 @@ impl<T: BeaconChainTypes> Service<T> {
     ) -> error::Result<(Arc<Self>, mpsc::UnboundedSender<NetworkMessage>)> {
         // build the network channel
         let (network_send, network_recv) = mpsc::unbounded_channel::<NetworkMessage>();
-        // Get a reference to the beacon chain store
-        let store = beacon_chain.store.clone();
         // launch message handler thread
+        let store = beacon_chain.store.clone();
         let message_handler_send = MessageHandler::spawn(
             beacon_chain,
             network_send.clone(),
             executor,
             network_log.clone(),
         )?;
 
+        let propagation_percentage = config.propagation_percentage;
         // launch libp2p service
-        let libp2p_service = Arc::new(Mutex::new(LibP2PService::new(
-            config.clone(),
-            network_log.clone(),
-        )?));
+        let (network_globals, mut libp2p_service) =
+            LibP2PService::new(config, network_log.clone())?;
 
-        // Load DHT from store
-        let key = Hash256::from_slice(&DHT_DB_KEY.as_bytes());
-        let enrs: Vec<Enr> = match store.get(&key) {
-            Ok(Some(p)) => {
-                let p: PersistedDht = p;
-                p.enrs
-            }
-            _ => Vec::new(),
-        };
-        for enr in enrs {
-            libp2p_service.lock().swarm.add_enr(enr);
+        for enr in load_dht::<T>(store.clone()) {
+            libp2p_service.swarm.add_enr(enr);
         }
 
-        let libp2p_exit = spawn_service(
-            libp2p_service.clone(),
+        // a delay used to initialise code after the network has started
```

Comment:
This comment should be a little more descriptive as to why we need the delay imo